packages/syft/src/syft/core/tensor/autodp/single_entity_phi.py

```diff
@@ -376,55 +377,57 @@ def __repr__(self) -> str:
             f"{self.__class__.__name__}(entity={self.entity.name}, child={self.child})"
         )
 
+    # Check for shape1 = (1,s), and shape2 = (,s) --> as an example
     def __eq__(self, other: SupportedChainType) -> SingleEntityPhiTensor:
-
-        if is_acceptable_simple_type(other) or self.child.shape == other.child.shape:  # type: ignore
-            # if the tensor being compared is also private
-            if isinstance(other, SingleEntityPhiTensor):
```

Comment:
Any reason this was deleted?

Reply:
This would fail if self.child and other.child have shapes that are not equal, but still broadcastable, so I rewrote them to check for broadcastability, and if self.entity == other.entity:

[Link to relevant lines](https://github.com/OpenMined/PySyft/blob/a13862e62d75dd1cc3e3f2c64625f9abfb757320/packages/syft/src/syft/core/tensor/autodp/single_entity_phi.py#L381)